src/test/java/com/google/sps/MealServletTest.java

```diff
@@ -0,0 +1,140 @@
+import com.google.gson.Gson;
+import org.springframework.mock.web.MockHttpServletRequest;
+import org.springframework.mock.web.MockHttpServletResponse;
+
+public class MealServletTest{
+    private final LocalServiceTestHelper helper =
+        new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig());
+    private final Meal meal_0 = new Meal(
+        1L, "Fried potato", "Fried potato with mushrooms and onion.",
+            new ArrayList<>(Arrays.asList("potato", "onion", "mushrooms", "oil")), "Main");
+    private final Meal meal_1 = new Meal(
+        1L, "Vegetable soup", "Vegetable soup with onion.",
+            new ArrayList<>(Arrays.asList("potato", "onion", "cabbage", "mushrooms", "water", "carrot", "pumpkin")), "Soup");
+    private final Meal meal_2 = new Meal(
+        2L, "Chocolate cake", "Chocolate cake with butter cream and strawberry.",
+            new ArrayList<>(Arrays.asList("flour", "water", "butter", "cocoa powder", "chocolate", "sugar", "strawberry", "eggs")), "Dessert");
+    private final Meal meal_3 = new Meal(
+        3L, "", "", new ArrayList<>(), "");
+
+
+    @Before
+    public void setUp() {
+        helper.setUp();
+    }
+
+    @After
+    public void tearDown() {
+        helper.tearDown();
+    }
+
+    @Test
+    public void datastoreTest() {
+        DatastoreService ds = DatastoreServiceFactory.getDatastoreService();
+        ds.put(new Entity("Meal"));
+        ds.put(new Entity("Meal"));
+        assertEquals(2, ds.prepare(new Query("Meal")).countEntities());
+    }
+
+    @Test
+    public void getMealByIdTest() throws IOException, ServletException {
+        DatastoreService ds = DatastoreServiceFactory.getDatastoreService();
+        ds.put(createMealEntity(meal_2));
+        ds.put(createMealEntity(meal_1));
+        
+        MealServlet servlet = new MealServlet();
+        MockHttpServletRequest request = new MockHttpServletRequest("GET", "/meal/1");
+        MockHttpServletResponse response = new MockHttpServletResponse();
+
+        try {
+            servlet.doGet(request, response);
+        } catch(Exception e) {
+        }
+        Gson gson = new Gson();
+        String expected = gson.toJson(meal_2);
+        String actual = response.getContentAsString().trim();
+
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    public void getMealListTest() throws IOException, ServletException {
+        DatastoreService ds = DatastoreServiceFactory.getDatastoreService();
+        ds.put(createMealEntity(meal_2));
+        ds.put(createMealEntity(meal_1));
+        
+        MealServlet servlet = new MealServlet();
+        MockHttpServletRequest request = new MockHttpServletRequest("GET", "/meal");
+        MockHttpServletResponse response = new MockHttpServletResponse();
+
+        try {
+            servlet.doGet(request, response);
+        } catch(Exception e) {
+        }
+        HashMap<Long, Meal> meals = new HashMap();
+        meals.put(0L, meal_1);
```

Comment:
There's an assumption that keys and values are related, namely `key == value.id`. This relation doesn't hold here, because `meal_1.id == 1`, but it's assigned to key `0`.

Let's try to maintain the assumption in test data and assign `meal_1` to key `1L`, and `meal_2` -> `2L`.

Note, that if you decide to drop the HashMap and use a List in the result (as suggested by Bartek above), the problem will disappear.